src/sidebar/components/moderation/ModerationControl.tsx

```diff
@@ -0,0 +1,51 @@
+import { ModerationStatusSelect } from '@hypothesis/annotation-ui';
+
+import type { Annotation } from '../../../types/api';
+import { useSidebarStore } from '../../store';
+import ModerationStatusBadge from './ModerationStatusBadge';
+
+export type ModerationControlProps = {
+  annotation: Annotation;
+  /** Classes to apply to the ModerationStatusBadge in case it's rendered */
+  badgeClasses?: string | string[];
+};
+
+/**
+ * Render the proper moderation control/s for provided annotation.
+ * - ModerationStatusSelect for moderators of pre-moderated groups.
+ * - ModerationStatusBadge for own annotations in any group
+ */
+export default function ModerationControl({
+  annotation,
+  badgeClasses,
+}: ModerationControlProps) {
+  const store = useSidebarStore();
+  const currentUserId = store.profile().userid;
+  const groupIsPreModerated = store.focusedGroup()?.pre_moderated;
+  const currentUserIsModerator = annotation.actions?.includes('moderate');
```

Comment:
```suggestion
  const canModerate = annotation.actions?.includes('moderate');
```

The client doesn't know if the user is a "moderator", just that they can moderate this annotation.